packages/server/src/automations/tests/utilities/AutomationBuilder.ts

```diff
@@ -0,0 +1,136 @@
+import { v4 as uuidv4 } from "uuid"
+import { testAutomation } from "../../../api/routes/tests/utilities/TestFunctions"
+import {
+  RowCreatedTriggerInputs,
+  RowCreatedTriggerOutputs,
+} from "../../triggerInfo/rowSaved"
+import {
+  RowUpdatedTriggerInputs,
+  RowUpdatedTriggerOutputs,
+} from "../../triggerInfo/rowUpdated"
+import {} from "../../steps/createRow"
+import { BUILTIN_ACTION_DEFINITIONS } from "../../actions"
+import { TRIGGER_DEFINITIONS } from "../../triggers"
+import {
+  RowDeletedTriggerInputs,
+  RowDeletedTriggerOutputs,
+} from "../../triggerInfo/rowDeleted"
+import {
+  AutomationStepSchema,
+  AutomationTriggerSchema,
+  LoopStepInputs,
+  DeleteRowStepInputs,
+  UpdateRowStepInputs,
+  CreateRowStepInputs,
+} from "@budibase/types"
+import {} from "../../steps/loop"
+
+class AutomationBuilder {
+  private automationConfig: any = {
```

Comment:
Do we need these `any`s?

Reply:
You can get around a few of them by creating union types of all of the inputs and outputs.

```typescript
type AutomationStepInput = serverLogStepInput | n8nStepInput | ...
type AutomationStepOutput = serverLogStepOutput | n8nStepOutput | ...
```